management/connection.go

```diff
@@ -526,6 +529,41 @@ func (c *ConnectionOptionsOIDC) SetScopes(enable bool, scopes ...string) {
 	c.Scope = &scope
 }
 
+type ConnectionOptionsOAuth2 struct {
+	ClientID              *string `json:"client_id,omitempty"`
+	ClientSecret          *string `json:"client_secret,omitempty"`
+	AuthorizationEndpoint *string `json:"authorization_endpoint"`
+	TokenEndpoint         *string `json:"token_endpoint"`
+	Scope                 *string `json:"scope,omitempty"`
+
+	// Scripts for the connection
+	// Allowed keys are: "fetchUserProfile"
+	CustomScripts map[string]interface{} `json:"scripts,omitempty"`
```

Comment:
Why did you go for a `CustomScripts` here instead of `Scripts` as in the JSON tag?

Reply:
I chose this approach to keep the terraform schema as it is in the other cases. I think it would be nice to keep the same name for the same functionality, and then just change the json to satisfy the API.  
I can change it if you want

Reply:
Yeah, I see what you mean... But I think I would prefer to remain closer to the Auth0 API naming and make no assumptions on our end.

Reply:
I've changed to follow the API names.